### PR TITLE
Create validation flow and action bar

### PR DESCRIPTION
**Pull request by rememberlenny**
# What

The front-end now sends the form params to the backend for validation. After the validation check, the front-end does a form `submit`. This allows any errors from validation to be properly handled on the front-end, without accidentally updating the `dirrty` fields or getting inconsistent notifications

screen shot 2016-07-15 at 8 31 08 pm

### CODE REVIEW

File: app/assets/javascripts/details/shell/action_bar_bridge.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+var ActionBarBridge;
```
Comment (dogweather) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: This creates a global `ActionBarBridge` variable, doesn't it? Should this instead simply assign `window.ActionBarBridge` to the result of the IIFE?

File: app/assets/javascripts/details/shell/action_bar_bridge.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      notifications:  '#action-bar-status',
+      editMode:       '#mode-parent',
+      modalCard:      '#modal-wrapper',
+      updateView:     '#mode-parent'
+    }
+    this._overrideTestConfig(config);
+    this._blastOff();
+  }
+
+  ActionBarBridge.prototype._blastOff = function(){
```
Comment (dogweather) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: The name `_blastOff` isn't too clear about what this does — would you mind renaming it?

File: app/assets/javascripts/details/shell/action_bar_bridge.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+var ActionBarBridge;
+ActionBarBridge = (function() {
+
+  function ActionBarBridge(c2, config){
```
Comment (dogweather) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: Could you add some docs; what this class's responsibilities are, and why we have it?

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 2 above)
Comment (rememberlenny) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @dogweather This is the consistent name for the `init` method. Its being used in 4 other files, so I think its okay. 

@nickbristow @phirefly thoughts?

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 1 above)
Comment (rememberlenny) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @dogweather This is the naming convention we are using. Its consistent with the `coffeescript` output for creating constructors.

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 3 above)
Comment (rememberlenny) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @dogweather Yep! Will do.

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 2 above)
Comment (dogweather) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: if it's the `init` method, could we just call it `init` then? Smells of cowboy coding, IMO ;-)

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 1 above)
Comment (dogweather) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @rememberlenny think of my point of view: a skilled developer who's unfamiliar with the framework you've created. If this was angular, there'd be docs online that'd tell me the purpose of the class and point to examples how to use it. But here, we don't have that. So for the sake of someone new to the code, what does the ActionBarBridge do?

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 2 above)
Comment (rememberlenny) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @dogweather Its a standard from the modular `backbone.js` library `human.js`.

Link: https://github.com/HenrikJoreteg/humanjs

I didnt come up with the syntax. 

Theres a whole book outlining the writing style and naming pattern: http://read.humanjavascript.com/

File: app/assets/javascripts/details/shell/action_bar_bridge.js (hunk 1 above)
Comment (rememberlenny) [inline app/assets/javascripts/details/shell/action_bar_bridge.js]: @dogweather I agree with the window/global overkill. I was doing this intentionally to make these files testable while running the beta testing. The scope can be removed, but it helps with debugging.

These files will eventually be in the asset pipeline. When we make that switch, the plan is to change how the files are invoked.

**Comment by nickbristow** [comment]
@dogweather did you want to weigh in on this again before we merge?

**Comment by dogweather** [comment]
@nickbristow in my opinion, its not ready to be merged because of the issues i raised.  but if it works for you, thats cool.

**Comment by nickbristow** [comment]
Merging, we'll open a new PR to address some of the JS issues effecting the wider project.

**Comment by dogweather** [comment]
👍  @rememberlenny  @nickbristow Sounds good, thanks!

**Pull request merged**